python/sparkdl/graph/utils.py

```diff
@@ -89,77 +87,81 @@ def get_tensor(graph, tfobj_or_name):
         name = tfobj_or_name.name
     if not isinstance(name, six.string_types):
         raise TypeError('invalid tensor request for {} of {}'.format(name, type(name)))
-    _tensor_name = as_tensor_name(name)
+    _tensor_name = tensor_name(name, graph=None)
     tnsr = graph.get_tensor_by_name(_tensor_name)
     assert tnsr is not None, \
         'cannot locate tensor {} in current graph'.format(_tensor_name)
     return tnsr
 
-def as_tensor_name(name):
+def tensor_name(tfobj_or_name, graph=None):
     """
     Derive tf.Tensor name from an op/tensor name.
-    We do not check if the tensor exist (as no graph parameter is passed in).
+    If the input is a name, we do not check if the tensor exist
+    (as no graph parameter is passed in).
 
-    :param name: op name or tensor name
-    """
-    assert isinstance(name, six.string_types)
-    name_parts = name.split(":")
-    assert len(name_parts) <= 2, name_parts
-    if len(name_parts) < 2:
-        name += ":0"
-    return name
-
-def as_op_name(name):
-    """
-    Derive tf.Operation name from an op/tensor name
-    We do not check if the operation exist (as no graph parameter is passed in).
-
-    :param name: op name or tensor name
-    """
-    assert isinstance(name, six.string_types)
-    name_parts = name.split(":")
-    assert len(name_parts) <= 2, name_parts
-    return name_parts[0]
-
-def op_name(graph, tfobj_or_name):
-    """
-    Get the name of a tf.Operation
-
-    :param graph: tf.Graph, a TensorFlow Graph object
     :param tfobj_or_name: either a tf.Tensor, tf.Operation or a name to either
     """
-    graph = validated_graph(graph)
-    return get_op(graph, tfobj_or_name).name
+    # If `graph` is provided, directly get the graph operation
+    if graph is not None:
+        return get_tensor(tfobj_or_name, graph).name
+    # If `graph` is absent, check if other cases
+    if isinstance(tfobj_or_name, six.string_types):
+        # If input is a string, assume it is a name and infer the corresponding tensor name.
+        # WARNING: this depends on TensorFlow's tensor naming convention
+        name = tfobj_or_name
+        name_parts = name.split(":")
+        assert len(name_parts) <= 2, name_parts
+        if len(name_parts) < 2:
+            name += ":0"
+        return name
+    elif hasattr(tfobj_or_name, 'graph'):
+        return get_tensor(tfobj_or_name, tfobj_or_name.graph).name
+    else:
+        raise TypeError('invalid tf.Tensor name query type {}'.format(type(tfobj_or_name)))
 
-def tensor_name(graph, tfobj_or_name):
+def op_name(tfobj_or_name, graph=None):
     """
-    Get the name of a tf.Tensor
+    Derive tf.Operation name from an op/tensor name.
+    If the input is a name, we do not check if the operation exist
```

Comment:
"If the input is a TensorFlow object, or the graph is given, we also check that the tensor exists in the associated graph."